kvstore/local_kv_store.go

```diff
@@ -0,0 +1,512 @@
+				ProtocolID:   protocol,
+				KeyID:        key,
+				Counterparty: wallet.Counterparty{Type: wallet.CounterpartyTypeSelf},
+			},
+			Plaintext: valueBytes,
+		}
+		encryptResult, err := kv.wallet.Encrypt(ctx, encryptArgs, kv.originator)
+		if err != nil {
+			return "", fmt.Errorf("for key %s: %w", key, err)
+		}
+		valueBytes = encryptResult.Ciphertext
+	}
+
+	pushDrop := pushdroptpl.PushDropTemplate{Wallet: kv.wallet, Originator: kv.originator}
+	lockingScript, err := pushDrop.Lock(
+		ctx,
+		[][]byte{valueBytes},
+		kv.getProtocol(key),
+		key,
+		wallet.Counterparty{Type: wallet.CounterpartyTypeSelf},
+		true,
+		false,
+		true,
+	)
+	if err != nil {
+		// Error creating script itself, not a wallet op yet
+		return "", fmt.Errorf("failed to create PushDrop locking script: %w", err)
+	}
+
+	inputs := make([]wallet.CreateActionInput, 0, len(lookupResult.outpoints))
+	for _, outpoint := range lookupResult.outpoints {
+		unlocker := pushDrop.Unlock(
+			ctx,
+			kv.getProtocol(key),
+			key,
+			wallet.Counterparty{Type: wallet.CounterpartyTypeSelf},
+			wallet.SignOutputsAll,
+			false,
+		)
+		inputs = append(inputs, wallet.CreateActionInput{
+			Outpoint:              outpoint,
+			InputDescription:      "Previous key-value token",
+			UnlockingScriptLength: unlocker.EstimateLength(),
+		})
+	}
+
+	createArgs := wallet.CreateActionArgs{
+		Description: fmt.Sprintf("Update %s in %s", key, kv.context),
+		InputBEEF:   lookupResult.inputBeef,
+		Inputs:      inputs,
+		Outputs: []wallet.CreateActionOutput{
+			{
+				LockingScript:     lockingScript.String(),
+				Satoshis:          1,
+				OutputDescription: "Key-value token",
+				Basket:            kv.context,
+				Tags:              []string{key},
+			},
+		},
+		Options: &wallet.CreateActionOptions{
+			// RandomizeOutputs: false, // Default is usually false?
+		},
+	}
+
+	createResult, err := kv.wallet.CreateAction(ctx, createArgs, kv.originator)
+	if err != nil {
+		// Wrap wallet create action error
+		return "", fmt.Errorf("error CreateAction: %w", err)
+	}
+
+	if len(inputs) == 0 {
+		if createResult.Txid == "" {
+			return "", errors.New("CreateAction returned no txid and no signable transaction for new key")
+		}
+		return fmt.Sprintf("%s.0", createResult.Txid), nil
+	}
+
+	if createResult.SignableTransaction == nil {
+		return "", errors.New("CreateAction did not return a signable transaction when inputs were provided")
+	}
+
+	spends, err := kv.prepareSpends(ctx, key, inputs, createResult.SignableTransaction.Tx, createArgs.InputBEEF)
+	if err != nil {
+		// Error occurred during signing preparation (e.g., decoding tx, generating unlock script)
+		return "", fmt.Errorf("failed to prepare spends for signing: %w", err)
+	}
+
+	signArgs := wallet.SignActionArgs{
+		Reference: createResult.SignableTransaction.Reference,
+		Spends:    spends,
+	}
+
+	signResult, err := kv.wallet.SignAction(ctx, signArgs, kv.originator)
+	if err != nil {
+		// *** Add Relinquish logic here ***
```

Comment:
should it be `TODO` comment, and addressed later, or it's already addressed below (I suppose this is true)